src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -508,27 +507,27 @@ public void addOnRankedMatchNotificationListener(Consumer<MatchmakerInfoMessage>
               gameLaunchMessage.getArgs().add("/players " + gameLaunchMessage.getExpectedPlayers());
               gameLaunchMessage.getArgs().add("/startspot " + gameLaunchMessage.getMapPosition());
 
-              startGame(gameLaunchMessage, faction, RatingMode.LADDER_1V1);
+              startGame(gameLaunchMessage, faction, ratingMode);
             }))
         .exceptionally(throwable -> {
           if (throwable instanceof CancellationException) {
-            log.info("Ranked1v1 search has been cancelled");
+            log.info("Ranked search has been cancelled");
           } else {
-            log.warn("Ranked1v1 could not be started", throwable);
+            log.warn("Ranked could not be started", throwable);
           }
           return null;
         });
   }
 
-  public void stopSearchLadder1v1() {
-    if (searching1v1.get()) {
-      fafService.stopSearchingRanked();
-      searching1v1.set(false);
+  public void stopSearchLadder(RatingMode ratingMode) {
```

Comment:
I think I'd take a String argument `queueName` here. (I know the message field is called `mod`, it's legacy stuff that we're stuck with :P)

Reply:
We don't need to pass a queue name here. I passed RatingMode object because it has information about the technical name of the mod (which is required to send a stop-message to a server). If we replace RatingMode with a QueueType object (which should be dynamically received from the server), I will pass QueueType instead of RatingMode. Look at this answer: https://github.com/FAForever/downlords-faf-client/pull/1786#discussion_r443147988